mitmproxy/proxy/layers/quic.py

```diff
@@ -1121,6 +1123,7 @@ class ClientQuicLayer(QuicLayer):
 
     server_tls_available: bool
     """Indicates whether the parent layer is a ServerQuicLayer."""
+    recv_buffer: bytearray
```

Comment:
Shouldn't this much rather be `list[bytes]` and we invoke `quic.receive_datagram` repeatedly? For UDP I think it's important that we preserve message boundaries. aioquic happens to tolerate this, but that's not necessarily true for other QUIC implementations.

(fundamentally this also needs a fix in the NextLayer logic - but maybe not in this PR)